src/platform/utilities/api/index.js

```diff
@@ -1,8 +1,68 @@
 import * as Sentry from '@sentry/browser';
 
 import environment from '../environment';
+import ENVIRONMENTS from '../../../site/constants/environments';
 import localStorage from '../storage/localStorage';
 
+let sessionTimeoutMinutesLengthSSO = 0;
+const SSOeEndpoint = () => {
+  let environmentPrefix;
+  switch (ENVIRONMENTS.BUILDTYPE) {
+    case ENVIRONMENTS.VAGOVSTAGING:
+      environmentPrefix = 'sqa.';
+      break;
+    case ENVIRONMENTS.VAGOVDEV:
+      environmentPrefix = 'int.';
+      break;
+    default:
+      environmentPrefix = '';
+  }
+
+  return `https://${environmentPrefix}eauth.va.gov/keepalive`;
+};
+
+function checkAndUpdateSSOeSession() {
+  const sessionExpiration = localStorage.getItem('sessionExpirationSSO');
+
+  if (sessionExpiration) {
+    const minutesUntilExpiration =
+      (sessionExpiration.getTime() - Date.now()) / (1000 * 60);
+
+    if (minutesUntilExpiration / sessionTimeoutMinutesLengthSSO < 0.67) {
+      // We want to minimize external keepalive calls, so this functions
+      // as self-enforced rate limiting
+      return;
+    }
+  }
+
+  fetch(SSOeEndpoint(), {
+    method: 'GET',
+    credentials: 'include',
+    cache: 'no-store',
+  })
+    .then(res => {
+      const hasSSOsession = res.headers.get('session-alive');
+
+      if (hasSSOsession === 'true') {
+        const sessionTimeout = res.headers.get('session-timeout');
+        sessionTimeoutMinutesLengthSSO = sessionTimeout / 60;
```

Comment:
I think something like `ssoSessionLengthInMinutes` would read better to me. Easier to reason around this logic when it's in terms of a session length or duration rather than a "session timeout".

Reply:
In fact, it doesn't seem like it's necessary to convert to minutes, since it's only done because the `minutesUntilExpiration` was declared in terms of minutes. It seems most useful to have everything in milliseconds.

```
// Session timeout is in seconds. Convert to milliseconds.
ssoSessionLength = res.headers.get('session-timeout') * 1000;
```

It avoids doing any conversions in setting the new `sessionExpirationSSO` and in the rate limiting code above too. Also, calling `minutesUntilExpiration` something like `remainingSessionTime` might help the condition read pretty intuitively.

```
const percentageOfSessionLengthThreshold = 0.67;
const remainingSessionTime = sessionExpiration.getTime() - Date.now();

if (remainingSessionTime / ssoSessionLength < percentageOfSessionLengthThreshold) {
  ...
}
```

Reply:
This a lot of rephrasing, and I might be misinterpreting what's going on. Just a suggestion to see if it makes sense to you.

Reply:
no, that makes a lot of sense, thanks for the suggestions. after thinking over it again more closely, I like minimizing the date conversions back and forth. ultimately it just leads to more confusion.

keeping things in milliseconds in more streamlined and plays nicely with using a percentage threshold